Purchases/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -186,6 +190,52 @@ public extension StoreProduct {
 
 }
 
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+public extension StoreProduct {
+    /// Asynchronously determines the subset of ``discounts`` that's eligible for the current user.
+    /// - Parameter purchases: the instance of `Purchases` to use when determining eligibility.
+    /// - Seealso: ``discounts``
+    /// - Note: if checking for eligibility for a `StoreProductDiscount` fails (for example, if network is down),
+    /// that discount will fail silently and be considered not eligible.
+    func eligibleDiscounts(withPurchases purchases: Purchases) async -> [StoreProductDiscount] {
```

Comment:
Is passing in a `Purchases` reference a new public API concept? I

It seems like might expect an API that is something like 👇 

```swift
Purchases.getEligibleDiscounts(forStoreProduct: storeProduct)
```

Thoughts? 😊

Reply:
I agree it's slightly weird, but it improves discoverability. When users type `product.discounts` they'll see the autocomplete for both.
What do you think?

Reply:
🤔 +1 to Josh's point, I would have expected the API to look exactly like what he described. 
That said, I do see the advantage of having access to it from `discounts`. 

Could we have both? That way, if you're used to using `Purchases` as the entrypoint, it won't feel weird, but you can also use the convenience of getting the eligible discounts easily. 

Alternatively, we could go ahead and use the `shared` instance of `Purchases` ourselves, since that's the one we're supposed to be using in the first place. 

Note that in either case we do have to think about what happens if this is somehow called with a non-configured instance of `Purchases`

Reply:
> Alternatively, we could go ahead and use the shared instance of Purchases ourselves, since that's the one we're supposed to be using in the first place.
> Note that in either case we do have to think about what happens if this is somehow called with a non-configured instance of Purchases

Oooh of course! I don't like implicitly depending on a global, but in the case of this SDK it's basically unusable without calling `configure`.
I'll also add a note to indicate that it's required 👍

Reply:
Done.

Reply:
Awesome! ❤️ 

@NachoSoto any thoughts on this bit? 

> Could we have both? That way, if you're used to using Purchases as the entrypoint, it won't feel weird, but you can also use the convenience of getting the eligible discounts easily.

Reply:
Oh yes of course! Done!